File: 20221005/모범답안/20221005_09/MatrixHasOutline_test.js
import assert from 'assert'
import solution from './MatrixHasOutline.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const matrix = [[true, true, true, true], [true, true, false, true], [true, false, true, true], [true, true, true, true]]

    // when
    const result = solution(matrix)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const matrix = [[true, true], [true, true]]

    // when
    const result = solution(matrix)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const matrix = [[true]]

    // when
    const result = solution(matrix)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const matrix = [[true, false], [true, true]]

    // when
    const result = solution(matrix)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const matrix = [[false]]

    // when
    const result = solution(matrix)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const matrix = [[false, true, false], [true, false, true], [false, true, false]]

    // when
    const result = solution(matrix)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });
})

File: 20221005/모범답안/20221005_09/MatrixHasOutline_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.medium.MatrixHasOutline.MatrixHasOutline import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        matrix = [[True, True, True, True], [True, True, False, True], [True, False, True, True], [True, True, True, True]]

        # when
        result = solution(matrix)

        # then
        self.assertEqual(True, result)

    def test_2(self):
        # given
        matrix = [[True, True], [True, True]]

        # when
        result = solution(matrix)

        # then
        self.assertEqual(True, result)

    def test_3(self):
        # given
        matrix = [[True]]

        # when
        result = solution(matrix)

        # then
        self.assertEqual(True, result)

    def test_4(self):
        # given
        matrix = [[True, False], [True, True]]

        # when
        result = solution(matrix)

        # then
        self.assertEqual(False, result)

    def test_5(self):
        # given
        matrix = [[False]]

        # when
        result = solution(matrix)

        # then
        self.assertEqual(False, result)

    def test_6(self):
        # given
        matrix = [[False, True, False], [True, False, True], [False, True, False]]

        # when
        result = solution(matrix)

        # then
        self.assertEqual(False, result)


if __name__ == '__main__':
    unittest.main()
